File: test/render.js
const fs = require("fs");
const should = require("should");
const flam3 = require("../lib/index.js");
const { PNG } = require("pngjs");
const pngdiff = require("png-diff");
const path = require("path");

let test_dir = path.join(path.dirname(flam3.path), "tests");
let data_dir = path.join(__dirname, "data");

describe("render", () => {
  it("test0 should render correctly", function (done) {
    let xml = fs.readFileSync(path.join(data_dir, "test0.flam3"), { encoding: "UTF8" });
    let genomes = flam3.Genome.fromXMLString(xml, "test0.flam3");
    should(genomes).have.length(1);
    let genome = genomes[0];

    genome.render({ seed: "testseed" }, function(err, data) {
      should(err).be.null();

      let png = new PNG({ width: genome.width, height: genome.height });
      png.data = data;

      pngdiff.outputDiffStream(png.pack(), path.join(test_dir, "test0.png"), function(err, stream, diffState) {
        should(err).be.null();
        should(diffState).equal(0);
        done();
      });
    });
  });

  it("test1 should render correctly", function (done) {
    let xml = fs.readFileSync(path.join(data_dir, "test1.flam3"), { encoding: "UTF8" });
    let genomes = flam3.Genome.fromXMLString(xml, "test1.flam3");
    should(genomes).have.length(1);
    let genome = genomes[0];

    genome.render({ seed: "testseed" }, function(err, data) {
      should(err).be.null();

      let png = new PNG({ width: genome.width, height: genome.height });
      png.data = data;

      pngdiff.outputDiffStream(png.pack(), path.join(test_dir, "test1.png"), function(err, stream, diffState) {
        should(err).be.null();
        should(diffState).equal(0);
        done();
      });
    });
  });
});

File: src/render.py
import os
import sys
import subprocess

if (len(sys.argv) != 4):
    sys.exit(1)

def render(flame, output):
    env = {
        "isaac_seed": "testseed",
        "seed": "testseed",
        "in": flame,
        "flam3_palettes": os.path.join(sys.argv[1], "flam3-palettes.xml"),
        "out": output
    }

    subprocess.check_call([os.path.join(sys.argv[1], "flam3-render")], env = env)

bindir = sys.argv[1]
sourcedir = sys.argv[2]
outputdir = sys.argv[3]

try:
    os.mkdir(testdir)
except:
    pass

render(os.path.join(sourcedir, "test0.flam3"), os.path.join(outputdir, "test0.png"))
render(os.path.join(sourcedir, "test1.flam3"), os.path.join(outputdir, "test1.png"))
